libs/util/git/src/providers/aws/aws-code-commit.service.ts

```diff
@@ -60,81 +68,185 @@ export class AwsCodeCommitService implements GitProvider {
+      return {
+        admin: false,
+        defaultBranch: repositoryMetadata.defaultBranch,
+        fullName: repositoryMetadata.repositoryName,
+        name: repositoryMetadata.repositoryName,
+        private: true,
+        url: repositoryMetadata.cloneUrlHttp,
+        groupName: null,
+      };
+    } else {
+      throw new Error(`Repository ${repositoryName} not found`);
+    }
   }
+
   async getRepositories(
     getRepositoriesArgs: GetRepositoriesArgs
   ): Promise<RemoteGitRepos> {
-    throw new Error("Method not implemented.");
+    const { pagination } = getRepositoriesArgs;
+    const command = new ListRepositoriesCommand({
+      sortBy: "repositoryName",
+      order: "ascending",
+    });
+
+    // TODO improve pagination
+    const { repositories } = await this.awsClient.send(command);
+    let paginatedRepositories: RemoteGitRepository[] = [];
+    if (repositories && repositories.length > 0) {
+      const skip = (pagination.page - 1) * pagination.perPage;
+      const take = skip + pagination.perPage;
+      paginatedRepositories = repositories
+        .slice(skip, take)
+        .map((repository) => {
+          if (this.isRequiredValid(repository)) {
+            return {
+              admin: false,
+              defaultBranch: "",
+              fullName: repository.repositoryName,
+              name: repository.repositoryName,
+              private: true,
+              url: "",
+              groupName: null,
+            };
+          } else {
+            throw new Error(
+              `Repository ${repository.repositoryName} not found`
+            );
+          }
+        });
+    }
+    return {
+      pagination,
+      repos: paginatedRepositories,
+      total: repositories?.length || 0,
+    };
   }
+
   async createRepository(
     createRepositoryArgs: CreateRepositoryArgs
   ): Promise<RemoteGitRepository | null> {
-    throw new Error("Method not implemented.");
+    const { repositoryName } = createRepositoryArgs;
+    const command = new CreateRepositoryCommand({
+      repositoryName,
+      repositoryDescription: "Created by Amplication",
+      tags: { system: "Amplication" },
+    });
+
+    const { repositoryMetadata } = await this.awsClient.send(command);
+
+    if (this.isRequiredValid(repositoryMetadata)) {
+      return {
+        admin: false,
+        defaultBranch: repositoryMetadata.defaultBranch,
+        fullName: repositoryMetadata.repositoryName,
+        name: repositoryMetadata.repositoryName,
+        private: true,
+        url: repositoryMetadata.cloneUrlHttp,
+        groupName: null,
+      };
+    } else {
+      throw new Error(`Repository ${repositoryName} not found`);
+    }
   }
+
   async deleteGitOrganization(): Promise<boolean> {
-    throw new Error("Method not implemented.");
+    // There is nothing to uninstall/delete when an organisation is deleted in AWS CodeCommit.
+    return true;
   }
+
   async getOrganization(): Promise<RemoteGitOrganization> {
```

Comment:
should this method be async?

Reply:
It's not an error since it returns a promise but I agree, it is irrelevant.